File: backend/tramonto/src/main/java/br/com/wsss/tramonto/dto/input/TestDto.java
package br.com.wsss.tramonto.dto.input;

import java.util.Date;
import java.util.HashSet;
import java.util.Set;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonView;

import br.com.wsss.tramonto.dto.output.View;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class TestDto extends BaseDto {

	@JsonView(View.Default.class)
	private String identifier;
	
	@JsonFormat(pattern="yyyy-MM-dd'T'HH:mm:ss")
	private Date initialDate;
	
	@JsonFormat(pattern="yyyy-MM-dd'T'HH:mm:ss")
	private Date finalDate;
	
	@JsonFormat(pattern="HH:mm:ss")
	private Date estimatedTime;
	
	private boolean retest;
	
	@JsonFormat(pattern="yyyy-MM-dd")
	private Date retestDate;
	
	@JsonView({View.Pagination.class})
	private String title;
	
	private String description;
	private String approach;
	private String aggression;
	private String generalObservation;
	private TestTypeDto type;
	private Set<ChecklistDto> checklists = new HashSet<>();
	private Set<TestObjectiveDto> objectives = new HashSet<>();
	private Set<TestVectorDto> vectors = new HashSet<>();
	private Set<StrategyDto> strategies = new HashSet<>();
	private Set<UserDto> testers = new HashSet<>();
	private boolean belongsToCurrentUser;

	
	@Getter
	@Setter
	public class TestDtoPage {}
	
}

File: frontend/tramonto-security/src/services/dtos/Test.dto.ts
import { Checklist } from 'stores/dtos/Checklist.dto';
import { StrategyDto } from 'src/services/dtos/StrategyInput.dto';
import { ToolDto } from 'src/services/dtos/Tool.dto';
import { TestObjective } from 'stores/dtos/TestObjective.dto';
import { TestVector } from 'components/test/Test';
import { UserDto } from 'stores/dtos/UserDto';

export default interface TestDto {
  id: string;
  title: string;
  description: string;
  initialDate: string;
  finalDate: string;
  estimatedTime: string;
  retest: boolean;
  retestDate: string;
  type: string;
  approach: string;
  aggression: string;
  generalObservation: string;
  belongsToCurrentUser: boolean;
  checklists: Array<Checklist>;
  strategies: Array<StrategyDto>;
  tools: Array<ToolDto>;
  vectors: Array<TestVector>;
  objectives: Array<TestObjective>;
  testers: Array<UserDto>;
  formatInitialAndFinalDate(): string;
};
